pcdsdevices/state.py

```diff
@@ -629,19 +714,27 @@ def state_config_dotted_names(state_count: int) -> list[str | None]:
     ----------
     state_count : int
         The number of known states used by the device.
+    motor_count : int, optional
+        The number of motors associated with the states config.
 
     Returns
     -------
     dotted_names : list of str or None
-        The full dotted names in state enum order.
+        The full dotted names in state enum order, going through all the states
+        of each motor before moving onto the next.
     """
-    return [None] + [
-        f'config.{get_dynamic_state_attr(num)}.state_name'
-        for num in range(1, state_count + 1)
-    ]
+    return [None] + state_config_dotted_attribute(
+        config_attr="state_name",
+        state_count=state_count,
+        motor_count=motor_count,
+        first_motor_only=True,
+    )
 
 
-def state_config_dotted_velos(state_count: int) -> list[str | None]:
+def state_config_dotted_velos(
+    state_count: int,
+    motor_count: int = 1,
+) -> list[str | None]:
```

Comment:
Also just `list[str]`, I think?

Reply:
in line 726 there's always `None` in the list at index 0.  I do wonder why this is necessary though..  (I'm sure it works it just seems weird)

Reply:
The diff makes this tougher to read than it should be...

Reply:
I need re-read these and double-check on index 0 and on the other one above

Reply:
These must have been copy/paste errors from the other function that has a None in the list.
I'm going to evaluate that None right now but I've fixed these two type annotations